pkg/plugins/config/config.go

```diff
@@ -52,57 +30,27 @@ type Cfg struct {
-	grafanaVersion string, logDatasourceRequests bool, pluginsCDNURLTemplate string, appURL string, appSubURL string, tracing Tracing, features featuremgmt.FeatureToggles, angularSupportEnabled bool,
-	grafanaComURL string, disablePlugins []string, hideAngularDeprecation []string, forwardHostEnvVars []string, concurrentQueryCount int, azureAuthEnabled bool,
-	userFacingDefaultError string, dataProxyRowLimit int64,
-	sqlDatasourceMaxOpenConnsDefault int, sqlDatasourceMaxIdleConnsDefault int, sqlDatasourceMaxConnLifetimeDefault int,
-) *Cfg {
-	return &Cfg{
-		log:                                 log.New("plugin.cfg"),
-		PluginsPath:                         pluginsPath,
-		BuildVersion:                        grafanaVersion,
-		DevMode:                             devMode,
-		PluginSettings:                      pluginSettings,
-		PluginsAllowUnsigned:                pluginsAllowUnsigned,
-		DisablePlugins:                      disablePlugins,
-		AWSAllowedAuthProviders:             awsAllowedAuthProviders,
-		AWSAssumeRoleEnabled:                awsAssumeRoleEnabled,
-		AWSExternalId:                       awsExternalId,
-		AWSSessionDuration:                  awsSessionDuration,
-		AWSListMetricsPageLimit:             awsListMetricsPageLimit,
-		AWSForwardSettingsPlugins:           AWSForwardSettingsPlugins,
-		Azure:                               azure,
-		ProxySettings:                       secureSocksDSProxy,
-		LogDatasourceRequests:               logDatasourceRequests,
-		PluginsCDNURLTemplate:               pluginsCDNURLTemplate,
-		Tracing:                             tracing,
-		GrafanaComURL:                       grafanaComURL,
-		GrafanaAppURL:                       appURL,
-		GrafanaAppSubURL:                    appSubURL,
-		Features:                            features,
-		AngularSupportEnabled:               angularSupportEnabled,
-		HideAngularDeprecation:              hideAngularDeprecation,
-		ForwardHostEnvVars:                  forwardHostEnvVars,
-		ConcurrentQueryCount:                concurrentQueryCount,
-		AzureAuthEnabled:                    azureAuthEnabled,
-		UserFacingDefaultError:              userFacingDefaultError,
-		DataProxyRowLimit:                   dataProxyRowLimit,
-		SQLDatasourceMaxOpenConnsDefault:    sqlDatasourceMaxOpenConnsDefault,
-		SQLDatasourceMaxIdleConnsDefault:    sqlDatasourceMaxIdleConnsDefault,
-		SQLDatasourceMaxConnLifetimeDefault: sqlDatasourceMaxConnLifetimeDefault,
+// NewPluginsCfg returns a new PluginsCfg.
+func NewPluginsCfg(devMode bool, pluginsPath string, pluginSettings setting.PluginSettings, pluginsAllowUnsigned []string,
+	pluginsCDNURLTemplate string, appURL string, appSubURL string, tracing Tracing, features featuremgmt.FeatureToggles,
+	angularSupportEnabled bool, grafanaComURL string, disablePlugins []string, hideAngularDeprecation []string,
+	forwardHostEnvVars []string) *PluginsCfg {
```

Comment:
have you thought of passing here `settings.Cfg` as a parameter to avoid the big list of parameters?